src/Sylius/Behat/Context/Domain/ProductContext.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Domain;
+
+use Behat\Behat\Context\Context;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+final class ProductContext implements Context
+{
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $productRepository;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $variantRepository;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $reviewRepository;
+
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param RepositoryInterface $productRepository
+     * @param RepositoryInterface $variantRepository
+     * @param RepositoryInterface $reviewRepository
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        RepositoryInterface $productRepository,
+        RepositoryInterface $variantRepository,
+        RepositoryInterface $reviewRepository
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->productRepository = $productRepository;
+        $this->variantRepository = $variantRepository;
+        $this->reviewRepository = $reviewRepository;
+    }
+
+    /**
+     * @When I delete the :productName product
+     */
+    public function iDeleteTheProduct($productName)
+    {
+        $product = $this->productRepository->findOneBy(['name' => $productName]);
+
+        if (null === $product) {
+            throw new \InvalidArgumentException(sprintf('Product with %s name was not found in the product repository', $productName));
+        }
+
+        $this->sharedStorage->set('deleted_product', $product);
+
+        $this->productRepository->remove($product);
+    }
+
+    /**
+     * @Then there should be no :variantName variant of this product in the product catalog
+     */
+    public function thereIsNoProductVariant($variantName)
+    {
+        expect($this->variantRepository->findOneBy(['presentation' => $variantName]))->toBe(null);
```

Comment:
This logic is implemented in [ProductTransformer](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Behat/Context/Transform/ProductContext.php#L60) We can think about implementing another transformer or just copy-paste this logic